controller/controller.go

```diff
@@ -1201,6 +1217,31 @@ func (ctrl *ProvisionController) updateDeleteStats(volume *v1.PersistentVolume,
 	}
 }
 
+// rescheduleProvisioning signal back to the scheduler to retry dynamic provisioning
+// by removing the annSelectedNode annotation
+func (ctrl *ProvisionController) rescheduleProvisioning(claim *v1.PersistentVolumeClaim) error {
+	if _, ok := claim.Annotations[annSelectedNode]; !ok {
+		// Provisioning not triggered by the scheduler, skip
+		return nil
+	}
+
+	// The claim from method args can be pointing to watcher cache. We must not
+	// modify these, therefore create a copy.
+	newClaim := claim.DeepCopy()
+	delete(newClaim.Annotations, annSelectedNode)
+	// Try to update the PVC object
+	if _, err := ctrl.client.CoreV1().PersistentVolumeClaims(newClaim.Namespace).Update(newClaim); err != nil {
+		return fmt.Errorf("delete annotation 'annSelectedNode' for PersistentVolumeClaim %q: %v", claimToClaimKey(newClaim), err)
+	}
+
+	// Save updated claim into informer cache to avoid operations on old claim.
+	if err := ctrl.claimInformer.GetStore().Update(newClaim); err != nil {
+		return fmt.Errorf("update claim informer cache for PersistentVolumeClaim %q: %v", claimToClaimKey(newClaim), err)
```

Comment:
I'm not happy with error handling here. We have updated the PVC in the apiserver, so it will get rescheduled eventually. But returning an error here is the same for the caller as if removing the label had failed and the next iteration will try to provision again instead of waiting for the scheduler.

This is a local operation. It shouldn't fail. Would it perhaps be better to panic when getting such an unexpected error?

Reply:
https://github.com/kubernetes/client-go/blob/eb0bad8167df60e402297b26e2cee1bddffde108/tools/cache/store.go#L154-L162 only fails when no key can be computed for the object, which should never happen.

Reply:
I'd log the error with Warning or something, the informer will get updated object from API server shortly. No need to throw an error.

Reply:
Done.